src/mqt/predictor/rl/predictorenv.py

```diff
@@ -351,6 +356,9 @@ def determine_valid_actions_for_state(self) -> list[int]:
         check_mapping = CheckMap(coupling_map=CouplingMap(self.device.coupling_map))
         check_mapping(self.state)
         mapped = check_mapping.property_set["is_swap_mapped"]
+        directionality_mapped = check_directionality(self.state, self.device.coupling_map)
+        if mapped and not directionality_mapped:  # there is an error in the
+            self.layout = None
```

Comment:
As per the comment in the helper, the code here can be simplified.

> If dag is mapped and the direction is correct the property is_direction_mapped is set to True (or to False otherwise).

So the above CheckMap can actually just be replaced by CheckGateDirection

Reply:
Although I am using the new check now, we actually need both checks since the their outcomes have different consequences: If we have a wrong directionality and a valid mapping, we often end in infinity loops since the layout is already fixed and the BasicSwap pass always leads to the same and wrong result. Therefore, in the case of a mapping with a wrong directionality, we want to undo the layout. 

If we only check the directionality, we would not be able to detect the case when a layout is set but no routing pass is applied yet.

Reply:
But how can there ever be a mapping with the wrong directionality and a layout being set?
Shouldn't we only employ routing passes that can handle directionality in case a device requires that?

Alternatively, we might add another property similar to how qiskit handles this. One is "is_swap_mapped", the other is "is_directionally_mapped" (or whatever these are called). In the last stage only optimizations that preserve the layout and connectivity are permitted.

Reply:
I added a function that now checks for unidirectionality which is a requirement for the RL predictor as of now.